manifests/package.pp

```diff
@@ -12,9 +12,18 @@
     }
   } else {
     package { $forge_server::package:
-      ensure   => present,
+      ensure   => $::forge_server::pkg_ensure,
       provider => $::forge_server::provider
     }
   }
 
+  # If we are ensuring the latest gem package, uninstall old versions
+  if $::forge_server::pkg_ensure == 'latest' {
```

Comment:
Personally I'm not really fond of implicit features. Also it would be kind of inconsistent if one ensures specific version.
Maybe it would be better if something like ```$cleanup = true``` would trigger the cleanup.